lm20/pipelines.py

```diff
@@ -168,22 +168,37 @@ def file_path(self, request, response=None, info=None, *, item=None):
 
         media_guid = hashlib.sha1(to_bytes(request.url)).hexdigest()
 
-        content_disposition = headers.get("Content-Disposition").decode()
-        _, params = cgi.parse_header(content_disposition)
-        filename = params["filename"]
+        content_disposition = headers.get("Content-Disposition")
+
+        if type(content_disposition) is bytes:
+            content_disposition = content_disposition.decode()
+
+        if content_disposition:
+            _, params = cgi.parse_header(content_disposition)
+            filename = params["filename"]
+
+            media_ext = os.path.splitext(filename)[1]
+        else:
+            filename = ""
```

Comment:
why do we need to set filename here?

Reply:
this was being used when looking for the `media_type` below in the `mimetypes.guess_type()` method.

but looking at this again, maybe i should just refactor that whole bit to be contained in this `if content_disposition` conditional. do you think something like this make sense (along with the other suggestions you've got here)?

```suggestion
        if content_disposition:
            _, params = cgi.parse_header(content_disposition)
            filename = params["filename"]
            media_ext = os.path.splitext(filename)[1]

            if media_ext not in mimetypes.types_map:
                media_ext = ""
                media_type = mimetypes.guess_type(filename)[0]
                if media_type:
                    media_ext = mimetypes.guess_extension(media_type)
                elif headers.get("Content-Type"):
                    content_type = headers["Content-Type"]
                    if type(content_type) is bytes:
                        media_ext = mimetypes.guess_extension(
                            content_type.decode("utf").split(";")[0]
                        )
                    else:
                        media_ext = mimetypes.guess_extension(
                            content_type.split(";")[0]
                        )
        else:
            media_ext = ""
```